docs/mockwbemserver.rst

```diff
@@ -836,389 +926,310 @@ mock environment.
+      See `Python operation methods in user-defined providers`_ for details.
 
-   .. autoautosummary:: pywbem_mock.FakedWBEMConnection
-      :attributes:
+      Example for an ``CreateInstance`` method of an instance write provider
+      that just calls superclass method to perform the work (and that could
+      therefore be omitted):
 
-   .. rubric:: Details
+      .. code-block:: python
 
-.. _`DMTF CIM schema download support`:
+          def CreateInstance(self, namespace, NewInstance):
+              return super(MyInstanceWriteProvider, self).CreateInstance(
+                  namespace, NewInstance)
 
-DMTF CIM schema download support
---------------------------------
+   d. Optional: It may define a post register setup method.
 
-.. automodule:: pywbem_mock._dmtf_cim_schema
+      The provider may override the
+      :meth:`~pywbem_mock.InstanceWriteProvider.post_register_setup` method
+      of the provider superclass to do any special setup it needs. That method
+      includes the current connection as a parameter so that WBEM operations
+      on the same or on different classes can be executed.
+      That method will be called during invocation of
+      :meth:`~pywbem_mock.FakedWBEMConnection.register_provider`, after the
+      provider registration is successful.
 
-.. autoclass:: pywbem_mock.DMTFCIMSchema
-   :members:
+      Example:
 
-   .. rubric:: Methods
+      .. code-block:: python
 
-   .. autoautosummary:: pywbem_mock.DMTFCIMSchema
-      :methods:
-      :nosignatures:
+          def post_register_setup(self, conn):
+            # code that performs post registration setup for the provider
 
-   .. rubric:: Attributes
+2. Register the user-defined provider using
+   :meth:`pywbem_mock.FakedWBEMConnection.register_provider`.
 
-   .. autoautosummary:: pywbem_mock.DMTFCIMSchema
-      :attributes:
+   This specifies the CIM namespaces for which the user-defined provider will
+   be active. These namespaces must already exist in the CIM repository
+   if the mock WBEM server.
 
-   .. rubric:: Details
+   .. code-block:: python
 
+       provider = MyInstanceProvider(self.cimrepository)
+       conn.register_provider(provider,
+                              namespaces=['root/interop', 'root/cimv2'])
 
 
-.. _`config`:
+.. _`Python operation methods in user-defined providers`:
 
-config
-------
+Python operation methods in user-defined providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. automodule:: pywbem_mock.config
-   :members:
+The *Python operation methods* (i.e. the Python methods implementing WBEM
+operations) in user-defined providers may:
 
+* provide parameter or CIM repository validation in addition to the normal
+  request validation,
+* modify parameters of the request,
+* abort the request with a :class:`pywbem.CIMError` exception,
+* make modifications to the CIM repository.
 
-.. _`Mock CIM repository`:
+The Python operation methods may or may not call the corresponding superclass
```

Comment:
Change may or may not call the corresponding superclass.  to "may call  the corresponding superclass method  to complete the CIM repository modification: or may implement the code to complete the modification.", Calling the corresponding superclass method insures that any validation that the superclass method is done (ex. insure that a CreateInstance does include all key properties) is done without duplicating code.

Reply:
DONE